network-config-analyzer/PoliciesFinder.py

```diff
@@ -0,0 +1,135 @@
+#
+# Copyright 2020- IBM Inc. All rights reserved
+# SPDX-License-Identifier: Apache2.0
+#
+
+from ruamel.yaml import YAML
+from collections import deque
+from bisect import insort
+from CmdlineRunner import CmdlineRunner
+from NetworkConfig import NetworkConfig, PoliciesContainer
+from NetworkPolicy import NetworkPolicy
+from K8sNetworkPolicy import K8sNetworkPolicy
+from CalicoNetworkPolicy import CalicoNetworkPolicy
+from IstioNetworkPolicy import IstioNetworkPolicy
+from K8sPolicyYamlParser import K8sPolicyYamlParser
+from CalicoPolicyYamlParser import CalicoPolicyYamlParser
+from IstioPolicyYamlParser import IstioPolicyYamlParser
+
+
+class PoliciesFinder:
+    """
+    This class is responsible for finding the network policies in the relevant input resources
+    The class contains several ways to build the set of policies (from cluster, from file-system, from GitHub).
+    """
+    def __init__(self):
+        self.policies_container = PoliciesContainer(policies={}, sorted_policies=[], profiles={}, allowed_labels=set())
```

Comment:
`allowed_labels` seems not relevant to `policies_conatiner` (it is not a policy).
Maybe it should be computed in the level of NetworkConfig,  after collecting all policies.
It is basically a set of strings computed as the union of each policy's `referenced_labels`.
I also think there is an issue in NetworkConfig when using `append_policy_to_config` and not taking into account the added policy's  `referenced_labels` within the config's `allowed_labels`. 
For `clone_without_policies ` the network config should not contain any allowed labels, as it has no policies with no referenced labels.